domain/schema/model.go

```diff
@@ -369,6 +369,44 @@ CREATE TABLE charm (
     uuid    TEXT PRIMARY KEY,
     url     TEXT NOT NULL
 );
+
+CREATE TABLE charm_storage (
+    charm_uuid       TEXT NOT NULL,
+    name             TEXT NOT NULL,
+    description      TEXT,
+    storage_kind_id  INT NOT NULL,
+    shared           BOOLEAN,
+    read_only        BOOLEAN,
+    count_min        INT NOT NULL,
+    count_max        INT NOT NULL,
+    minimum_size_mib INT,
+    location         TEXT,
+    CONSTRAINT       fk_storage_instance_kind
+        FOREIGN KEY  (storage_kind_id)
+        REFERENCES   storage_kind(id),
+    CONSTRAINT       fk_charm_storage_charm
+        FOREIGN KEY  (charm_uuid)
+        REFERENCES   charm(uuid),
+    PRIMARY KEY (charm_uuid, name)
+);
+
+CREATE INDEX idx_charm_storage_charm
+ON charm_storage (charm_uuid);
+
+CREATE TABLE charm_storage_property (
+    charm_uuid   TEXT NOT NULL,
+    storage_name TEXT NOT NULL,
+    property     TEXT NOT NULL,
+    CONSTRAINT       fk_charm_storage_property_charm_storage
+        FOREIGN KEY  (charm_uuid, storage_name)
+        REFERENCES   charm_storage(charm_uuid, name),
+    PRIMARY KEY (charm_uuid, storage_name, property)
```

Comment:
What is an example of a property value?

Reply:
The doc I found says
```
	// Properties allow the charm author to characterise the relative storage
	// performance requirements and sensitivities for each store.
	// eg “transient” is used to indicate that non persistent storage is acceptable,
	// such as tmpfs or ephemeral instance disks.
	//
	// Properties has no default, and is optional.
```
I didn't realise this existed

Reply:
Hmm. It's mentioned in https://juju.is/docs/sdk/storage. If it's not factored Juju-side, I bet no one is relying on it charm-side.

Reply:
I say we leave it until we know it's needed.